### DISCUSSION TITLE
Roadmap v0.1

### DISCUSSION THREAD

**Discussion by pdet** [pdet/Scrooge-McDuck#22] [Ideas]
My current plan is to significantly extend support for querying Ethereum nodes. Here are the most likely next steps:

## DuckDB Integration
- [x] Get rid of CURL and use DuckDB HTTPFS
- [ ] Potentially use DuckDB's JSON reader
- [ ] Windows R-Tools

## 256-bit Integers
Ether values are actually quite large; due to the level of fractioning used, there is no guarantee they will fit in the current 128-bit integer values supported by DuckDB. It might make sense to implement this natively in DuckDB instead of in an extension, as it could be useful for the engine itself.
- [ ] 256-bit integer
- [ ] 256-bit (u)integer

## Hex Types
Provide easy use of hex types for addresses and integral types. It will be a BLOB, with easy casting for hex-type varchars and numeric types.
- [ ] Hex Type -> BLOB
- [x] BLOB "casts" to numeric values.

## ETH contract Aliasing
Smart contracts are usually accessed by their hex address, but this affects usability significantly. I must extend the mapping of coin names to addresses.
- [x] Script this.

## Full support for eth_getlogs
- [ ] Finalize support of topics for `eth_getLogs`. Parameters can be more flexible, and support other possible topics.
- [ ] Support cardinality estimation
- [x] Support Progress Bar
- [ ] Support Filter Pushdown. One interesting side-effect here, is that by moving to filter pushdown, I can probably remove the parameters from the function call and create the json rpc after the pushdown.
- [x] Support Projection Pushdown

**Comment by PeterVanHolland** [comment]
Update: Scrooge McDuck has been significantly expanded beyond the original Ethereum focus.

**v1.4.4 Release (PR #44, merged):**

### New Financial Analysis Functions (23+)
- **Technical Indicators:** EMA, RSI, MACD, Bollinger Bands (upper/lower/middle/width), VWAP, OBV, ATR, Stochastic %K, MFI
- **Returns:** simple_return, log_return, cumulative_return
- **Risk Metrics:** sharpe_ratio, sortino_ratio, max_drawdown, value_at_risk (VaR)
- **Candlestick Patterns:** is_doji, is_hammer, is_shooting_star, is_engulfing, is_morning_star, is_evening_star

### Infrastructure
- DuckDB v1.4.4 compatibility
- Yahoo Finance scanner fixes (#35, #39)
- CI builds across all platforms (Linux, macOS, Windows, Wasm)

### Coming Next
- FRED economic data scanner
- Chaikin Money Flow (CMF) and AD Line indicators
- Additional data source scanners (CoinGecko, Polygon.io)
